cmake/modules/AddSwiftHostLibrary.cmake

```diff
@@ -0,0 +1,65 @@
+# This source file is part of the Swift.org open source project
+#
+# Copyright (c) 2014 - 2022 Apple Inc. and the Swift project authors
+# Licensed under Apache License v2.0 with Runtime Library Exception
+#
+# See http://swift.org/LICENSE.txt for license information
+# See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+
+# Add a new host library with the given name.
+function(add_swift_host_library name)
+  set(ASHL_SOURCES ${ARGN})
+
+  # Create the library target.
+  add_library(${name} ${ASHL_SOURCES})
+
+  # Add this to the list of exported targets.
+  set_property(GLOBAL APPEND PROPERTY SWIFTSYNTAX_EXPORTS ${name})
+
+  # Determine where Swift modules will be built and installed.
+  set(module_dir ${CMAKE_LIBRARY_OUTPUT_DIRECTORY})
+  set(module_base "${module_dir}/${name}.swiftmodule")
+  set(module_file "${module_base}/${SWIFT_MODULE_TRIPLE}.swiftmodule")
+  set(module_interface_file "${module_base}/${SWIFT_MODULE_TRIPLE}.swiftinterface")
```

Comment:
I think we don’t use `module_interface_file` anywhere. Is that intentional?

Reply:
It's a vestige of the failed attempt to enable library evolution. There's a potential miscompile in the very old Swift on CI.